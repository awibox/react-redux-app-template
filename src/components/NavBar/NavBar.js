import React, { Component } from 'react';
import { NavLink } from 'react-router-dom';
import classNames from 'classnames';
import { navigationItems } from 'config';
// Styles
import styles from './NavBar.scss';

export default class NavBar extends Component {
  constructor(props) {
    super(props);
    this.state = {
      openNavMenu: false,
    };
    this.toggleMobileMenu = this.toggleMobileMenu.bind(this);
  }

  toggleMobileMenu() {
    this.setState({ openNavMenu: !this.state.openNavMenu });
  }

  renderLinks() {
    return (
      <div className={styles.navMenu}>
        {navigationItems.map((item) => (
          <NavLink key={item.route} exact className={styles.navMenuItem}
                   activeClassName={styles.navMenuItemActive}
                   onClick={() => this.closeMenu()}
                   to={item.route}>
            <i className={`fa fa-${item.icon}`}/>
            <span className={styles.navMenuItemText}>{item.name}</span>
          </NavLink>
        ))}
      </div>
    );
  }

  closeMenu() {
    this.setState({ openNavMenu: false });
  }

  render() {
    return (
      <div className={styles.nav}>
        <div className={styles.navDesktopMenu}>
          {this.renderLinks()}
        </div>
        <div className={classNames(styles.navMobile,
          { [styles.navMobileActive]: this.state.openNavMenu })}
             onClick={this.toggleMobileMenu}>
          <i className="fa fa-list"></i>
        </div>
        <div className={classNames(styles.navMobileMenu,
          { [styles.show]: this.state.openNavMenu })}>
          {this.renderLinks()}
        </div>
      </div>
    );
  }
}
