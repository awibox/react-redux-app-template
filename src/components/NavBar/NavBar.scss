@import '../../styles/variables';

.nav {
  position: relative;
  display: flex;
  justify-content: flex-end;
}
.navDesktopMenu {
  display: block;
}
.navMobileMenu {
  display: none;
}
.navMenu {
  float: right;
  display: flex;
  flex-flow: row;
}
.navMenuItem {
  color: $headerTextColor;
  padding: 17px $defaultIndent;
  text-decoration: none;
  font-size: 16px;
  user-select: none;
  white-space: nowrap;
  &.navMenuItemActive {
    background: $navActiveBackground;
  }
  i, .navMenuItemText {
    margin: 0 3px;
  }
}

.navMobile {
  display: none;
  font-size: 22px;
  line-height: 22px;
  color: $headerTextColor;
  padding: 15px 20px 4px;
  cursor: pointer;
}
.navMobileActive {
  color: #555;
}
.show {
  display: block;
}
@media (min-width: $widthScreenTabletPortrait) {
  .nav {
    .navMobileMenu {
      display: none;
    }
  }
}
@media (max-width: $widthScreenTabletPortrait) {
  .navMobile {
    display: block;
  }
  .navDesktopMenu {
    display: none;
  }
  .navMobileMenu {
    position: absolute;
    top: 53px;
    right: 0;
    z-index: 99;
    .navMenu {
      flex-flow: column;
      background: #fff;
      box-shadow: -1px 1px 3px $headerBorderColor;
      .navMenuItem {
        border: 1px solid #e4eaec;
        padding: $defaultIndent $defaultIndent*2;
        color: #555555;
        &.navMenuItemActive {
          background: #eaebec;
        }
      }
    }
  }
}
