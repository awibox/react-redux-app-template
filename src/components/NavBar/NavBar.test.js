import React from 'react';
import { shallow } from 'enzyme';
import NavBar from './NavBar';

describe('NavBar', () => {
  it('should render correctly', () => {
    const component = shallow(<NavBar />);
    expect(component).toMatchSnapshot();
  });
  it('mobile menu should be close', () => {
    const component = shallow(<NavBar />);
    expect(component.state('openNavMenu')).toEqual(false);
  });
  it('should open mobile menu', () => {
    const component = shallow(<NavBar />);
    component.find('.navMobile').simulate('click');
    expect(component.state('openNavMenu')).toEqual(true);
  });
});
