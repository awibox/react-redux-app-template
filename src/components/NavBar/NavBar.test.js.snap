// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar should render correctly 1`] = `
<div
  className="nav"
>
  <div
    className="navDesktopMenu"
  >
    <div
      className="navMenu"
    >
      <NavLink
        activeClassName="navMenuItemActive"
        className="navMenuItem"
        exact={true}
        key="/"
        onClick={[Function]}
        to="/"
      >
        <i
          className="fa fa-home"
        />
        <span
          className="navMenuItemText"
        >
          Home page
        </span>
      </NavLink>
      <NavLink
        activeClassName="navMenuItemActive"
        className="navMenuItem"
        exact={true}
        key="/repos"
        onClick={[Function]}
        to="/repos"
      >
        <i
          className="fa fa-database"
        />
        <span
          className="navMenuItemText"
        >
          Repositories
        </span>
      </NavLink>
      <NavLink
        activeClassName="navMenuItemActive"
        className="navMenuItem"
        exact={true}
        key="/author"
        onClick={[Function]}
        to="/author"
      >
        <i
          className="fa fa-address-card"
        />
        <span
          className="navMenuItemText"
        >
          Author
        </span>
      </NavLink>
    </div>
  </div>
  <div
    className="navMobile"
    onClick={[Function]}
  >
    <i
      className="fa fa-list"
    />
  </div>
  <div
    className="navMobileMenu"
  >
    <div
      className="navMenu"
    >
      <NavLink
        activeClassName="navMenuItemActive"
        className="navMenuItem"
        exact={true}
        key="/"
        onClick={[Function]}
        to="/"
      >
        <i
          className="fa fa-home"
        />
        <span
          className="navMenuItemText"
        >
          Home page
        </span>
      </NavLink>
      <NavLink
        activeClassName="navMenuItemActive"
        className="navMenuItem"
        exact={true}
        key="/repos"
        onClick={[Function]}
        to="/repos"
      >
        <i
          className="fa fa-database"
        />
        <span
          className="navMenuItemText"
        >
          Repositories
        </span>
      </NavLink>
      <NavLink
        activeClassName="navMenuItemActive"
        className="navMenuItem"
        exact={true}
        key="/author"
        onClick={[Function]}
        to="/author"
      >
        <i
          className="fa fa-address-card"
        />
        <span
          className="navMenuItemText"
        >
          Author
        </span>
      </NavLink>
    </div>
  </div>
</div>
`;
